labrad/protocol.py

```diff
@@ -333,5 +373,92 @@ def __init__(self, source, context, target):
     def __repr__(self):
         return 'MessageContext(source=%s, ID=%s, target=%s)' % (self.source, self.ID, self.target)
 
-# factory for churning out LabRAD connections
+
+# factory for creating LabRAD connections
 factory = protocol.ClientCreator(reactor, LabradProtocol)
+
+
+@inlineCallbacks
+def connect(host=C.MANAGER_HOST, port=None, tls=C.MANAGER_TLS):
```

Comment:
It would be really nice to factor out the tls stuff into a subroutine. It's just a little bit hard to read through the code with all the tls stuff defined as an inlined function etc etc.